src/libcollections/dlist.rs

```diff
@@ -675,7 +675,7 @@ impl<A: fmt::Show> fmt::Show for DList<A> {
 }
 
 #[stable]
-impl<S: Writer, A: Hash<S>> Hash<S> for DList<A> {
+impl<S: Writer + Hasher, A: Hash<S>> Hash<S> for DList<A> {
```

Comment:
I do think that this is needed because of the `self.len().hash(state)` b/c `uint` requires `Writer`